tgui/packages/tgui/interfaces/PumpControl.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC (https://choosealicense.com/licenses/isc/)
+ */
+
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Section, Slider, Stack } from '../components';
+// import { DataInputOptions } from './common/DataInput';
+import { Window } from '../layouts';
+
+// List of information about a pump
+type PumpData = {
+  tag: string; // Pump name
+  netid: string; // Pump id
+  power: string; // On or off
+  target_output: number; // Current output target of the pump
+  min_output: number;
+  max_output: number;
+  area_name: string; // Name of the area this pump is in
+  processing: boolean; // Whether we are waiting for packet response or not
+  alive: number; // A value of -1, 0, or 1 where -1 is checking if the pump is alive, 0 is dead, and 1 is alive
+}
+
+// List of areas which have pumps
+type AreaList = {
+  area_list: { [key: string] : { [key: string] : PumpData } };
+  frequency: number;
+};
+
+// Responsible for providing information and settings for a pump.
+const PumpSettings = (props:any, context:any) => {
+  const { act } = useBackend<PumpData>(context);
+  const { pump } = props;
+  // Local states allow to keep the appearance of seamless response, but do not cope well with button spamming
+  const [target_output, setOutput] = useLocalState(context, pump.netid+"pressure", pump.target_output);
+  const [power, setPower] = useLocalState(context, pump.netid+"power", "on");
+
+  const setPressure = (newPressure: number) => {
+    setOutput(newPressure);
+    act('setPressure', { netid: pump.netid, pressure: newPressure });
+  };
+  const togglePump = () => {
+    setPower((power === "on") ? "off" : "on");
+    act('togglePump', { netid: pump.netid });
+  };
+
+  return (
+    <Box>
+      <Stack py={1}>
+        <Stack.Item>
+          {pump.tag}
+        </Stack.Item>
+        {
+          <Stack.Item textAlign={"right"} grow={1}>
+            {pump.alive === -1 ? "Establishing Connection..." : "Connected"}
+          </Stack.Item>
+        }
+      </Stack>
+      <Stack>
+        <Stack.Item>
+          <Button
+            width={4}
+            icon="power-off"
+            color={(power === 'on') ? "green" : "red"}
+            onClick={() => togglePump()}>
+            {(power === 'on') ? 'On' : 'Off'}
+          </Button>
+        </Stack.Item>
+        <Stack.Item grow>
+          <Slider
+            disabled={pump.alive !== 1}
+            value={target_output}
+            minValue={pump.min_output}
+            maxValue={pump.max_output}
+            unit={"kPa"}
+            stepPixelSize={0.05}
+            onChange={(_e: any, value: number) => setPressure(value)}
+          />
+        </Stack.Item>
+      </Stack>
+    </Box>
+  );
+};
+
+// Responsible for creating a section for the pumps in an area.
+const PumpArea = (_:any, context:any) => {
```

Comment:
_ means an unused variable, so this should use an actual name instead.

Reply:
renamed to props